packages/target-tauri/translationfn/src/lib.rs

```diff
@@ -0,0 +1,278 @@
+                        let mut msg = message.clone();
+                        for matches in self.var_finder.find_iter(message) {
+                            let from = matches.as_str();
+                            let position: u32 = from.chars().nth(1).unwrap().to_digit(10).unwrap();
+                            msg = msg.replace(from, items.get(position as usize - 1).unwrap());
+                        }
+                        msg
+                    } else {
+                        error!("Message not existing for {key}");
+                        key.to_owned()
+                    }
+                }
+                Substitution::Quantity(quantity) => {
+                    let quantity_key = match self.plural_rules.select(quantity) {
+                        Ok(PluralCategory::ZERO) => "zero",
+                        Ok(PluralCategory::ONE) => "one",
+                        Ok(PluralCategory::TWO) => "two",
+                        Ok(PluralCategory::FEW) => "few",
+                        Ok(PluralCategory::MANY) => "many",
+                        Ok(PluralCategory::OTHER) => "other",
+                        Err(_) => {
+                            error!("Failed to  {quantity}");
+                            return key.to_owned();
+                        }
+                    };
+                    if let Some(message) = entry.get(quantity_key) {
+                        message.replace("%d", &quantity.to_string())
+                    } else {
+                        error!(
+                            "Message for key was found {key}, but variant {quantity_key} to represent quantity of {quantity} is missing"
+                        );
+                        key.to_owned()
+                    }
+                }
+                Substitution::QuantityFloat(quantity_float) => {
+                    let quantity_key = match self.plural_rules.select(quantity_float) {
+                        Ok(PluralCategory::ZERO) => "zero",
+                        Ok(PluralCategory::ONE) => "one",
+                        Ok(PluralCategory::TWO) => "two",
+                        Ok(PluralCategory::FEW) => "few",
+                        Ok(PluralCategory::MANY) => "many",
+                        Ok(PluralCategory::OTHER) => "other",
+                        Err(_) => {
+                            error!("Failed to  {quantity_float}");
+                            return key.to_owned();
+                        }
+                    };
+                    if let Some(message) = entry.get(quantity_key) {
+                        message.replace("%d", &quantity_float.to_string())
+                    } else {
+                        error!(
+                            "Message for key was found {key}, but variant {quantity_key} to represent quantity of {quantity_float} is missing"
+                        );
+                        key.to_owned()
+                    }
+                }
```

Comment:
I believe this could be rewritten by making the function generic.
But this is also manageable as is.

Reply:
Let's mark as resolved since it's not important